### PR TITLE
CORE-10424 : Changes for managing session for interop processor

### PR DESCRIPTION
**Pull request by SaurabhR3**
1. Changes for managing session for interop processor
2. Added Interop State

### CODE REVIEW

**Review by yarivus — APPROVED** [review]
LGTM

**Review by ac101m — COMMENTED** [review]
Best bump the api revision to 3 for this!
Bear in mind that the api and runtime respositories are currently out of sync, if you have pending changes in runtime-os as well, you'll need to wait for CORE-10420 to be merged before merging them (blame szymon!).

**Review by jmacmahonr3 — APPROVED** [review]
LGTM

**Review by yarivus — APPROVED** [review]
LGTM

File: data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropState.avsc
Comment (szymonsztuka) [inline data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropState.avsc:32]: I think the best would be to store groupId as well, in a as separate field. StateProcessor would be completely independent from a message coming back from `FlowProcessor` in terms of deriving alias groupId (as it would store it previously) and no need to get groupId from the destination field.

Comment (SaurabhR3) [reply data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropState.avsc:32]: I think we don't need that right now. We can easily derive it from destination alias in case of outbound message. In future if we need this we can add it.

**Review by ac101m — CHANGES_REQUESTED** [review]
LGTM, but as my PR is still in progress, you should use version 2 rather than 3 as originally discussed.

File: data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropState.avsc
Comment (szymonsztuka) [reply data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropState.avsc:32]: I would not expect that the destination alias will have a correct alias group name (of the "returning" messaged from initated flow) , we haven't really finished implementation inside FlowProcessor. Also saving alias group ID here, allows us to store the full context , not a half of the context. It's better for Session management to be fully independent, and we can also detect any incompatibilities (e.g. bug in FlowProcessor) when a message comes back (crucial in early development stage). Also from admin point of view, imagine at later stage investigating a queue to troubleshooting etc - we can se any Identity names but we have no clue what groups they belong for.

**Review by szymonsztuka — CHANGES_REQUESTED** [review]

File: data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropState.avsc
Comment (SaurabhR3) [reply data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropState.avsc:32]: Fixed in 755153e

**Review by ac101m — APPROVED** [review]
LGTM

**Review by szymonsztuka — APPROVED** [review]

**Review by yarivus — APPROVED** [review]
LGTM

**Review by yarivus — APPROVED** [review]
LGTM

**Review by yarivus — APPROVED** [review]
LGTM

**Pull request merged**
